datafusion/datasource-json/src/source.rs

```diff
@@ -130,8 +149,20 @@ impl FileSource for JsonSource {
         Arc::new(conf)
     }
 
-    fn with_projection(&self, _config: &FileScanConfig) -> Arc<dyn FileSource> {
-        Arc::new(Self { ..self.clone() })
```

Comment:
It's unclear to me if this contains bugs or not. It seems like the projection was silently dropped on the floor...